### PR TITLE
feat(kernel): auto-record stage_runs from stage-transition comments (5a5ba3a6)

### PR DESCRIPTION
**Pull request by harshanandak**
## Summary

Completes the phase fix started in PR #348. #348 added the `stage_runs` record/read capability (`recordStageRun`, `getCurrentStage`, `listStageRuns`, the `forge stage` verb, and `current_stage` on `forge show`) but **nothing called it automatically**, so `stage_runs` stayed empty and `current_stage` was always unknown.

This wires it into normal use. The Descriptive Context Convention (AGENTS.md) already has agents record every stage boundary as a kernel comment shaped `stage: <from> -> <to>`. This PR turns that already-followed convention into a structured stage run: on a successful kernel `comment`, Forge best-effort **completes `<from>` and starts `<to>`** in `stage_runs`. No new manual verb call, no change to agent behavior — `current_stage` just becomes real.

## Integration point & why

The comment write path (`lib/commands/_issue.js`) — not `enforceStageEntry` — because Forge stages are agent-driven via skills; agents don't run `forge plan`/`forge dev` as CLI commands, but they DO already run `forge issue comment "stage: X -> Y ..."` at every stage exit (with the resolved issue id in hand). Hooking there makes recording automatic through the path agents already traverse.

## How a stage now records a run

`forge comment <id> "stage: dev -> validate\n..."` → after the comment lands, `recordStageTransitionFromComment` parses the `stage:` line and calls `driver.recordStageRun` complete(dev) + start(validate). `forge show <id>` then reports `current_stage: validate (active)`.

## Best-effort / non-blocking

`recordStageTransition` **never throws** — a `stage_run` write failure (parse miss, DB error, missing driver) can never fail the comment that triggered it. Kernel path only; reuses the driver already on `opts` (no second DB handle).

## Changes
- `lib/workflow/stage-transition.js` (new): `parseStageTransition` (pure) + `recordStageTransition` (best-effort).
- `lib/commands/_issue.js`: fire `recordStageTransitionFromComment` after a successful kernel comment.
- Tests: unit (parser + recorder edge cases) + end-to-end through `runIssueSubcommand`.

## Tests
- `test/workflow/stage-transition.test.js`, `test/commands/comment-stage-transition.test.js` — new, green.
- Existing stage/comment/issue suites green; `eslint --max-warnings 0` clean.

## Scope call / follow-up (5bfd2414)
Smallest version that makes phase real for the main stages via the transition comment. Deferred: (1) explicit `forge stage --start/--complete` in each canonical skill so the first stage's start and final stage's completion (no transition comment) are also recorded; (2) dashboard/status snapshot consuming `current_stage`; (3) the AGENTS.md doc note (protected `generated_harness` surface).

Issue: 5a5ba3a6-e3a9-4f55-a1c9-4707182c2989

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added the read-only `forge claims [--json]` command to list active issue leases.
  * Comments containing `stage: <from> -> <to>` now automatically record completion of the previous stage and start of the next stage when using the Kernel backend.
* **Bug Fixes**
  * Stage-recording failures no longer interrupt or invalidate successful comment submissions.
  * Invalid or non-stage comments are safely ignored.

### CODE REVIEW

File: lib/workflow/stage-transition.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      return { recorded: false };
+    }
+    driver.recordStageRun(
+      { issue_id: issueId, stage: transition.from, action: 'complete' },
+      config,
+    );
+    driver.recordStageRun(
+      { issue_id: issueId, stage: transition.to, action: 'start' },
+      config,
+    );
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply lib/workflow/stage-transition.js]: ✅ Fixed: transition write is now atomic.

Changed: Added a transactional `recordStageTransition({issue_id, from, to})` on the sqlite driver (lib/kernel/sqlite-driver.js) that wraps complete(from) + start(to) in ONE `BEGIN IMMEDIATE` transaction and ROLLBACKs on any failure. lib/workflow/stage-transition.js now calls that atomic op instead of two separate recordStageRun calls (a non-atomic fallback remains only for minimal drivers that lack the op). Caller stays best-effort/non-blocking.
Reason: If start(to) threw, the completed from-stage was already persisted — a half-transition with wrong current_stage. The transaction makes it all-or-nothing.
Test: test/kernel/stage-runs.test.js asserts that when the to-write fails mid-transaction the completed from-stage rolls back (dev stays active, no half-transition persisted).
Commit: ce0db0b2

File: lib/workflow/stage-transition.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+function parseStageTransition(body) {
+  if (typeof body !== 'string' || body.length === 0) {
+    return null;
+  }
+  const match = body.match(STAGE_LINE);
+  if (!match) {
+    return null;
+  }
+  const from = normalizeStageId(match[1]);
+  const to = normalizeStageId(match[2]);
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply lib/workflow/stage-transition.js]: ✅ Fixed: uppercase stage tokens now parse.

Changed: parseStageTransition (lib/workflow/stage-transition.js) lowercases both regex captures before normalizeStageId — `normalizeStageId(match[1].toLowerCase())` / `match[2].toLowerCase()`.
Reason: STAGE_LINE matches case-insensitively, so `stage: DEV -> VALIDATE` matched but was then rejected because normalizeStageId only knows lowercase canonical ids.
Test: test/workflow/stage-transition.test.js adds a case asserting `stage: DEV -> VALIDATE` → {from:'dev', to:'validate'}.
Commit: ce0db0b2
